### PR TITLE
MPI RTD 2.0

### PR DESCRIPTION
**Pull request by joseph-cooke**
Inspiration for update from https://www.notion.so/nuits/RTD-OpenMPI-c19e04cce2bd4551a1cbd18fd2c4e440

### CODE REVIEW

File: docs/source/software/mpi.md [hunk 1]
```diff
@@ -1,28 +1,325 @@
...
+The mpi4py package has been designed to be as close as possible to the MPI standard, providing Python developers with a familiar and straightforward interface to MPI.
+
+In this program, process 0 sends the number -1 to process 1, which receives it and prints it.
+
+To install mpi4py inside of a conda environment:
+
+:::{code-block} bash
+srun -n 4 --pty /bin/bash
+module load anaconda3/2022.05
+source activate <mpi4py_env>
```
Comment (SerenaCaplins) [inline docs/source/software/mpi.md:259]: Are you missing the conda create command here?

**Review by SerenaCaplins — APPROVED** [review]
Looks really good! I mostly read for readability etc. And found no major issues besides the conda create command.

File: docs/source/software/mpi.md (hunk 1 above)
Comment (joseph-cooke) [reply docs/source/software/mpi.md]: Good catch. I corrected that and pointed to using the prefix command as we do in the conda documentation. Thank you!!

**Review by visionjo — APPROVED** [review]
I made a few changes:
- Minimal language changes
- Added `dollarmath` extension to render latex $\pi$
- Changed inline notes to callouts
- Added links to the benchmarks
- Add dropdown to show the printout of module show openmpi

All looks good, @joseph-cooke , validate and merge!

**Pull request merged**